docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,292 @@
+                   request body, should be set.
+                   
+**Effects**:
+- _on success_:
+The file indicated by `/path/to/snapshot_file` contains the devices' model state and
+emulation state, same as when creating a full snapshot. The one indicated by
+`/path/to/mem_file` contains this time a **diff copy** of the guest memory - the diff
+consists of the memory pages which have been dirtied since the last snapshot creation
+or since the creation of the microVM, whichever of these events was the most recent.
+All the other effects mentioned in the **Effects** section from `## Creating full snapshots`
+apply here.
+- _on failure_: no side-effects.
+
+*Note*: This is an example of an API command that enables dirty page tracking:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i  \
+    -X PUT 'http://localhost/machine-config' \
+    -H 'Accept: application/json'            \
+    -H 'Content-Type: application/json'      \
+    -d '{
+            "vcpu_count": 2,
+            "mem_size_mib": 1024,
+            "ht_enabled": false,
+            "track_dirty_pages": true
+    }'
+```
+
+Enabling this support enables KVM dirty page tracking, so it comes at a cost
+(which consists of CPU cycles spent by KVM accounting for dirtied pages); it should only
+be used when needed.
+
+Creating a snapshot will **not** influence state, will **not** stop or end the microVM,
+it can be used as before, so the microVM can be resumed if you still want to use it.
+
+### Resuming the microVM
+
+You can resume the microVM by sending the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/vm" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Resumed"
+    }"
+```
+
+**Prerequisites**: The microVM is `Paused`.
+                   Successive calls of this request are ignored (microVM remains
+                   in the running state).
+**Effects**:
+- _on success_: microVM is guaranteed to be `Resumed`.
+- _on failure_: no side-effects.
+
+## Loading snapshots
+
+If you want to load a snapshot, you can do that only **before** the microVM is configured
+(the only resources that can be configured prior are the Logger and the Metrics systems)
+by sending the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/load" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+            \"mem_file_path\": \"/path/to/mem_file\",
+            \"enable_diff_snapshots\": true
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: A full memory snapshot and a microVM state file **must** be provided.
```

Comment:
What about adding disks, net ifaces, vsock backend management as a prerequisite as well?
Smth along the lines:

> Users must make sure _disk backing files, network interfaces backing TAPs and/or Vsock backend socket_ are set up and accessible to the new firecracker process where the microVM is resumed. These host-resources need to be accessible at the same relative paths to the new firecracker process as they were to the original firecracker process.

Reply:
Sounds good.